File: Year-2025 Problem List/March 2025/2206. Divide Array Into Equal Pairs/2206. Divide Array Into Equal Pairs.py
# Problem          : 2206. Divide Array Into Equal Pairs
# Time Complexity  : O(N)
# Space Complexity : O(N)
# Python Solution

class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        Hash = defaultdict(int)
        
        for num in nums:
            Hash[num] += 1
        
        for count in Hash.values():
            if count % 2 != 0:
                return False
        return True

File: Year-2025 Problem List/March 2025/2206. Divide Array Into Equal Pairs/2206. Divide Array Into Equal Pairs.cpp
/*
# Problem          : 2206. Divide Array Into Equal Pairs
# Time Complexity  : O(N)
# Space Complexity : O(N) 
# C++ Solution
*/

class Solution {
public:
    bool divideArray(vector<int>& nums) {
        map<int, int> mp;
        for (auto num : nums) {
            mp[num]++;
        }

        bool ok = true;
        for (auto [x, y] : mp) {
            if (y % 2) {
                ok = false;
                break;
            }
        }
        return ok;
    }
};